.aws/src/main.ts

```diff
@@ -28,32 +29,64 @@ class CuratedCorpusAPI extends TerraformStack {
     super(scope, name);
 
     new AwsProvider(this, 'aws', { region: 'us-east-1' });
-
-    new PagerdutyProvider(this, 'pagerduty_provider', { token: undefined });
     new LocalProvider(this, 'local_provider');
     new NullProvider(this, 'null_provider');
+    new PagerdutyProvider(this, 'pagerduty_provider', { token: undefined });
 
     new RemoteBackend(this, {
       hostname: 'app.terraform.io',
       organization: 'Pocket',
       workspaces: [{ prefix: `${config.name}-` }],
     });
 
-    const pocketVpc = new PocketVPC(this, 'pocket-vpc');
-    const region = new DataAwsRegion(this, 'region');
     const caller = new DataAwsCallerIdentity(this, 'caller');
+    const region = new DataAwsRegion(this, 'region');
+    const pocketVpc = new PocketVPC(this, 'pocket-vpc');
+    const curatedCorpusPagerduty = this.createPagerDuty();
 
     const pocketApp = this.createPocketAlbApplication({
       rds: this.createRds(pocketVpc),
       s3: this.createS3Bucket(),
-      pagerDuty: this.createPagerDuty(),
+      pagerDuty: curatedCorpusPagerduty,
       secretsManagerKmsAlias: this.getSecretsManagerKmsAlias(),
       snsTopic: this.getCodeDeploySnsTopic(),
       region,
       caller,
     });
 
     this.createApplicationCodePipeline(pocketApp);
+
+    new PocketAwsSyntheticChecks(this, 'synthetics', {
+      // alarmTopicArn: config.environment === 'Prod' ? curatedCorpusPagerduty.snsCriticalAlarmTopic.arn : '', // this should be improved, empty string recreates updates constantly as is in cdktf
+      alarmTopicArn: '', // remove this line & uncomment above line when we're ready to alert on synchecks
+      environment: process.env.NODE_ENV === 'development' ? 'Dev' : 'Prod', // yes we should use config.environment, but needs more refinment in module
+      prefix: config.prefix,
+      query: [
+        {
+          endpoint: config.domain,
+          data: '{"query": "query {scheduledSurface(id: \\"NEW_TAB_EN_US\\"){id}}"}', // New Tab relies upon scheduledSurface query
+          jmespath: 'data.scheduledSurface.id',
+          response: 'NEW_TAB_EN_US',
+        },
+        {
+          endpoint: config.domain,
+          data: '{"query": "query { scheduledSurface(id: \\"NEW_TAB_EN_US\\") {items(date: \\"2023-05-30\\") {corpusItem {id, url}}}}"}', // New Tab also relies upon corpusItem resolution
+          jmespath:
+            'to_string(length(data.scheduledSurface.items[*].corpusItem.id))',
+          response: `${config.environment === 'Prod' ? '25' : '2'}`,
+        },
```

Comment:
I would not hardcode the IDs expected here - rather just make sure they are returned and are of the correct type (e.g. integer) if that is possible with `jmespath`. For a thorough check, I'd also make sure _all_ fields are returned without an error - in the past, we've had issues with queries where querying for one specific field would make the whole query die with a 500 error, whereas the same query without that field include was fine.

Reply:
So per discussion last night with Nina:
1. these aren't hardcoded ids - the existing check is confirming the length of items returned matches what exists for this older date in dev & prod.
2. dev may fail, but we have no guarantees of uptime on dev, hence no alerting on dev, so that's fine - the dev synthetic check primarily exists to confirm the fitness of the check, not the uptime of the dev service.
3. there's a new feature request here for synthetics to do more than issue request confirm response via matching - there's a shape validation request, and possibly a smaller request to do more than matching - have variable comparison operators (so you can do greater than, less than etc). This isn't impossible, just more than synchecks defaults tend towards, so working on some approach for this atm.

Beyond that, its unclear there is a good way to test the fitness / health of specific "entity resolution" in our subgraphs without having known entities the synthetic can faithfully call to confirm. Ideally, we start to roll those out in prod so there are easier checks, but we'll see if the shape validation route can be done without adding much to the syncheck code itself, which should remain as simple as feasible.

Reply:
Alright, pushed up a change that will fail on this check if a non-empty list of corpusItems show up or any of these shows up without an id field.

we will want to expand eventually to cover more fields on a corpusItem that indicate health (or not) of the service, but need to balance operational health vs data validation / code validation, which are different vectors of alarm.